source/utilities.cc

```diff
@@ -104,6 +105,43 @@ namespace aspect
       return static_cast<bool>(ifile); // only in c++11 you can convert to bool directly
     }
 
+    void
+    load_and_distribute_file_content(const std::string &filename,
+                                     const MPI_Comm &comm,
+                                     std::stringstream &datastream)
+    {
+      // Reset datastream
+      datastream.str(std::string());
+
+      if (Utilities::MPI::this_mpi_process(comm) == 0)
+        {
+          std::ifstream filestream(filename);
+          AssertThrow (filestream,
+                       ExcMessage (std::string("Could not open file <") + filename + ">."));
+
+          // Read data from disk
+          filestream >> datastream.rdbuf();
```

Comment:
This is interesting. Does this really read the entire file in one statement? Regardless of how large it is? I don't think I've ever seen this :-)